rllib/agents/sac/rnnsac.py

```diff
@@ -47,16 +47,26 @@ def validate_config(self, config: TrainerConfigDict) -> None:
         # Call super's validation method.
         super().validate_config(config)
 
-        if config["replay_buffer_config"]["replay_sequence_length"] != -1:
+        # Add the `burn_in` to the Model's max_seq_len.
+        replay_sequence_length = (
+            config["replay_buffer_config"]["replay_burn_in"]
+            + config["model"]["max_seq_len"]
+        )
+        # Check if user tries to set replay_sequence_length (to anything
+        # other than the proper value)
+        if config["replay_buffer_config"]["replay_sequence_length"] not in [
+            -1,
+            replay_sequence_length,
+        ]:
             raise ValueError(
                 "`replay_sequence_length` is calculated automatically to be "
-                "model->max_seq_len + burn_in!"
+                "config['model']['max_seq_len'] + config['burn_in']. Leave "
+                "config['replay_sequence_length'] blank to avoid this error."
             )
-        # Add the `burn_in` to the Model's max_seq_len.
         # Set the replay sequence length to the max_seq_len of the model.
-        config["replay_buffer_config"]["replay_sequence_length"] = (
-            config["burn_in"] + config["model"]["max_seq_len"]
-        )
+        config["replay_buffer_config"][
+            "replay_sequence_length"
+        ] = replay_sequence_length
```

Comment:
@ArturNiederfahrenhorst @sven1977 I have a nit about this pattern that I am seeing a lot throughout RLlib's codebase.
I think we should treat config as a "const" obj. This is what users tell us, and likely what we should record with our checkpoints etc. modifying this in-place feels pretty confusing.
if we need to derive a control bit from some user configured inputs, we can create an internal only local configuration, but that should be part of the implementation details, not getting reflected back at the original cfg.